main/main.c

```diff
@@ -2282,6 +2286,9 @@ int php_module_startup(sapi_module_struct *sf, zend_module_entry *additional_mod
 		module->info_func = PHP_MINFO(php_core);
 	}
 
+	/* Extensions that add engine hooks after this point do so at their own peril */
+	zend_finalize_system_id();
```

Comment:
May be call zend_finalize_system_id() from opcache post_startup. I'm also not completely sure. Just an idea...

Reply:
If we moved it there, I'm not sure how we'd be able to finalize the system ID if OPcache is not installed.